stripe/src/main/java/com/stripe/android/paymentsheet/model/ClientSecret.kt

```diff
@@ -25,6 +33,11 @@ internal data class PaymentIntentClientSecret(
 ) : ClientSecret() {
     override fun createConfirmParamsFactory() =
         ConfirmPaymentIntentParamsFactory(this)
+
+    override suspend fun processPaymentFlowResultWithProcessor(
+        paymentFlowResult: PaymentFlowResult.Unvalidated,
+        processor: PaymentFlowResultProcessor
+    ) = processor.processPaymentIntent(paymentFlowResult)
```

Comment:
This pattern seems inverted. In my mind `ClientSecret` is just a typed String. Why would it know how to process a payment?

Reply:
Yes, I agree. It's a WIP, this gives the result that I want, but I'm trying to come up with a better pattern around having this clean logic and the right class names/functionalities. Happy to take any ideas you might have.